### PR TITLE
Fix installation on Windows runners

### LINKED ISSUE #8: Failing on Windows runners
When running on a Windows action runner with `runner: windows-latest`, this action fails on index.js line 36:

https://github.com/GoTestTools/gotestfmt-action/blob/65f1d2228f06cc5e828b84597440fbd063d12ea2/index.js#L36

<details>
<summary>
Full Workflow log
</summary>
```
Run GoTestTools/gotestfmt-action@v[2](https://github.com/norwd/pword/actions/runs/3341225601/jobs/5532217476#step:4:2).1.0
  with:
    token: ***
    org: GoTestTools
    repo: gotestfmt
  env:
    GOROOT: C:\hostedtoolcache\windows\go\1.19.2\x64
Running on OS 'msys_nt-10.0-20[3](https://github.com/norwd/pword/actions/runs/3341225601/jobs/5532217476#step:4:3)[4](https://github.com/norwd/pword/actions/runs/3341225601/jobs/5532217476#step:4:4)8'
Downloading latest stable gotestfmt version starting with v2. from GoTestTools/gotestfmt...
Found release v2.4.1 matching criteria, attempting to download binary...
Examining release asset gotestfmt_2.4.1_checksums.txt at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.1/gotestfmt_2.4.1_checksums.txt ...
Examining release asset gotestfmt_2.4.1_darwin_amd[6](https://github.com/norwd/pword/actions/runs/3341225601/jobs/5532217476#step:4:6)4.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.1/gotestfmt_2.4.1_darwin_amd64.tar.gz ...
Examining release asset gotestfmt_2.4.1_darwin_arm64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.1/gotestfmt_2.4.1_darwin_arm64.tar.gz ...
Examining release asset gotestfmt_2.4.1_linux_3[8](https://github.com/norwd/pword/actions/runs/3341225601/jobs/5532217476#step:4:9)6.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.1/gotestfmt_2.4.1_linux_386.tar.gz ...
Examining release asset gotestfmt_2.4.1_linux_amd64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.1/gotestfmt_2.4.1_linux_amd64.tar.gz ...
Examining release asset gotestfmt_2.4.1_linux_arm64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.1/gotestfmt_2.4.1_linux_arm64.tar.gz ...
Examining release asset gotestfmt_2.4.1_windows_386.zip at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.1/gotestfmt_2.4.1_windows_386.zip ...
Examining release asset gotestfmt_2.4.1_windows_amd64.zip at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.1/gotestfmt_2.4.1_windows_amd64.zip ...
Binary download failed, trying next release. (No release asset matched postfix '_msys_nt-[10](https://github.com/norwd/pword/actions/runs/3341225601/jobs/5532217476#step:4:11).0-[20](https://github.com/norwd/pword/actions/runs/3341225601/jobs/5532217476#step:4:21)348_amd64.tar.gz'.)
Found release v2.4.0 matching criteria, attempting to download binary...
Examining release asset gotestfmt_2.4.0_checksums.txt at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.0/gotestfmt_2.4.0_checksums.txt ...
Examining release asset gotestfmt_2.4.0_darwin_amd64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.0/gotestfmt_2.4.0_darwin_amd64.tar.gz ...
Examining release asset gotestfmt_2.4.0_darwin_arm64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.0/gotestfmt_2.4.0_darwin_arm64.tar.gz ...
Examining release asset gotestfmt_2.4.0_linux_386.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.0/gotestfmt_2.4.0_linux_386.tar.gz ...
Examining release asset gotestfmt_2.4.0_linux_amd64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.0/gotestfmt_2.4.0_linux_amd64.tar.gz ...
Examining release asset gotestfmt_2.4.0_linux_arm64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.0/gotestfmt_2.4.0_linux_arm64.tar.gz ...
Examining release asset gotestfmt_2.4.0_windows_386.zip at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.0/gotestfmt_2.4.0_windows_386.zip ...
Examining release asset gotestfmt_2.4.0_windows_amd64.zip at https://github.com/GoTestTools/gotestfmt/releases/download/v2.4.0/gotestfmt_2.4.0_windows_amd64.zip ...
Binary download failed, trying next release. (No release asset matched postfix '_msys_nt-10.0-20348_amd64.tar.gz'.)
Found release v2.3.2 matching criteria, attempting to download binary...
Examining release asset gotestfmt_2.3.2_checksums.txt at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.2/gotestfmt_2.3.2_checksums.txt ...
Examining release asset gotestfmt_2.3.2_darwin_amd64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.2/gotestfmt_2.3.2_darwin_amd64.tar.gz ...
Examining release asset gotestfmt_2.3.2_darwin_arm64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.2/gotestfmt_2.3.2_darwin_arm64.tar.gz ...
Examining release asset gotestfmt_2.3.2_linux_386.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.2/gotestfmt_2.3.2_linux_386.tar.gz ...
Examining release asset gotestfmt_2.3.2_linux_amd64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.2/gotestfmt_2.3.2_linux_amd64.tar.gz ...
Examining release asset gotestfmt_2.3.2_linux_arm64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.2/gotestfmt_2.3.2_linux_arm64.tar.gz ...
Examining release asset gotestfmt_2.3.2_windows_386.zip at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.2/gotestfmt_2.3.2_windows_386.zip ...
Examining release asset gotestfmt_2.3.2_windows_amd64.zip at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.2/gotestfmt_2.3.2_windows_amd64.zip ...
Binary download failed, trying next release. (No release asset matched postfix '_msys_nt-10.0-20[34](https://github.com/norwd/pword/actions/runs/3341225601/jobs/5532217476#step:4:35)8_amd64.tar.gz'.)
Found release v2.3.1: Bugfixes for go-acc matching criteria, attempting to download binary...
Examining release asset gotestfmt_2.3.1_checksums.txt at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.1/gotestfmt_2.3.1_checksums.txt ...
Examining release asset gotestfmt_2.3.1_darwin_amd64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.1/gotestfmt_2.3.1_darwin_amd64.tar.gz ...
Examining release asset gotestfmt_2.3.1_darwin_arm64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.1/gotestfmt_2.3.1_darwin_arm64.tar.gz ...
Examining release asset gotestfmt_2.3.1_linux_[38](https://github.com/norwd/pword/actions/runs/3341225601/jobs/5532217476#step:4:39)6.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.1/gotestfmt_2.3.1_linux_386.tar.gz ...
Examining release asset gotestfmt_2.3.1_linux_amd64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.1/gotestfmt_2.3.1_linux_amd64.tar.gz ...
Examining release asset gotestfmt_2.3.1_linux_arm64.tar.gz at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.1/gotestfmt_2.3.1_linux_arm64.tar.gz ...
Examining release asset gotestfmt_2.3.1_windows_386.zip at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.1/gotestfmt_2.3.1_windows_386.zip ...
Examining release asset gotestfmt_2.3.1_windows_amd64.zip at https://github.com/GoTestTools/gotestfmt/releases/download/v2.3.1/gotestfmt_2.3.1_windows_amd64.zip ...
Binary download failed, tried 4 times, giving up. (No release asset matched postfix '_msys_nt-10.0-203[48](https://github.com/norwd/pword/actions/runs/3341225601/jobs/5532217476#step:4:49)_amd64.tar.gz'.)
Setup failed.
Error: No release asset matched postfix '_msys_nt-10.0-20348_amd64.tar.gz'.
```
</details>

I thought this was possibly related to #4, but now my thinking is that there isn't a binary for windows in the latest release of `gotestfmt`?

### PR DESCRIPTION
**Pull request by norwd**
## Please describe the change you are making

* Add windows runner to the self test workflow
* Modify `determineOS` to detect windows correctly
* Modify `downloadRelease` to install the `gotestfmt` binary in an OS independent manner

## Your code will be released under [the MIT license](LICENSE.md). Are you in the position, and are you willing to release your code under this license?

Yup :)

Fixes #8

### CODE REVIEW

**Comment by norwd** [comment]
This took waaaaaay too many attempts :joy:

File: index.js [hunk 1]
```diff
@@ -9,31 +9,46 @@ async function downloadRelease(octokit, os, org, repo, release, token) {
...
-            console.log("Unpacking tar file...")
-            execSync("cd /usr/local/lib/gotestfmt && sudo tar -xvzf /tmp/gotestfmt.tar.gz")
-            console.log("Removing tarball...")
-            fs.unlinkSync("/tmp/gotestfmt.tar.gz")
-            console.log("Creating /usr/local/bin directory if it does not exist already...")
-            execSync("sudo mkdir -p /usr/local/bin")
-            console.log("Linking gotestfmt...")
-            execSync("sudo ln -s /usr/local/lib/gotestfmt/gotestfmt /usr/local/bin/gotestfmt")
+
+            // Extract the archive into the install target
```
Comment (engelmi) [inline index.js]: Minor: Please remove the inline comments. They are not needed in these cases as the code is self-explanatory. Also, over time comments tend to drift apart from the code leading to confusion.

**Comment by engelmi** [comment]
> This took waaaaaay too many attempts joy

But the end results looks really good :smiley:

File: index.js (hunk 1 above)
Comment (norwd) [reply index.js]: Fixed in [ef20a3a](https://github.com/GoTestTools/gotestfmt-action/pull/10/commits/ef20a3ae63b6c43b73c4811e1c7792c43d63a443)

**Pull request merged**